### PR TITLE
Simplify SX126x configuration & Add support for "E22-900M30S with ESP32-S3-WROOM-1-N4" board

### PR DESCRIPTION
**Pull request by S5NC**
Will require updating all variant files which use SX126x!
Will require adding board to [src/platform/esp32/architecture.h](https://github.com/meshtastic/firmware/blob/master/src/platform/esp32/architecture.h) and [src/mesh/generated/meshtastic/mesh.pb.h](https://github.com/meshtastic/firmware/blob/master/src/mesh/generated/meshtastic/mesh.pb.h)

Board name may need to be changed to something a little shorter, I trust you can find a name which meets current conventions, though please note that this is for the ESP32-**S3** in particular, the regular ESP32 would require slightly different files

`// FIXME`s added where relevant for future changes or improvements

Unsure whether the "-DARDUINO_ESP32S3_DEV" flag is needed, nor of its purpose, and the effect of the other extra flags in ESP32-S3-WROOM-1-N4.json

### CODE REVIEW

**Comment by jp-bennett** [comment]
Is this actually a board, or is it just the CPU? What is the specific hardware being targeted?

**Comment by jp-bennett** [comment]
I would suggest picking a name for the board, rather than just using the CPU name, to avoid confusion.

File: src/mesh/SX126xInterface.cpp [hunk 1]
```diff
@@ -26,20 +26,25 @@ template <typename T> bool SX126xInterface<T>::init()
...
+    LOG_DEBUG("DIO3_AS_TCXO_AT_1V8 mode false. Not using DIO3 to power TCXO\n");
+    float tcxoVoltage = 0; // None - we use an XTAL
 #endif
     bool useRegulatorLDO = false; // Seems to depend on the connection to pin 9/DCC_SW - if an inductor DCDC?
 
     RadioLibInterface::init();
 
-    if (power == 0)
-        power = SX126X_MAX_POWER;
+    // Do not correct 0dBm to 22dBm as may be using external amplifier! (removed code that did this)
```
Comment (thebentern) [inline src/mesh/SX126xInterface.cpp:44]: Don't make this change... we already have precdence for this. Look at the SX126X_MAX_POWER definition in the Station G1 variant for an example. Default limit for SX1262/8 is 22dbm. If targets have amplification, they must set their own limits in the variant

**Comment by thebentern** [comment]
> Will require updating all variant files which use SX126x! 
This doesn't sound too appealing, especially as we're trying to stabilize 2.2.x for a beta release.

File: src/mesh/SX126xInterface.cpp (hunk 1 above)
Comment (caveman99) [reply src/mesh/SX126xInterface.cpp:44]: That code was for setting sensible defaults when we install a new configuration cause of version changes. Not sure it's still neccessary, it sure was transitioning from 1.2 to 2.0

**Pull request closed without merging**